### DISCUSSION TITLE
Nextjs On Cpanel

### DISCUSSION THREAD

**Discussion by hajpepe** [vercel/next.js#39385] [Help]
Hello 
I tried to run the NextJS project on the cPanel host, but I get an error :
![Screenshot 2022-08-06 222758](https://user-images.githubusercontent.com/58140037/183301905-870d23bb-acf7-4b74-ad58-50b19f1c185a.png)
my server.js :
```
const { createServer } = require('http')
const { parse } = require('url')
const next = require('next')

const dev = process.env.NODE_ENV !== 'production'
const hostname = 'localhost'
const port =  process.env.PORT || 3022

const app = next({ dev, hostname, port })
const handle = app.getRequestHandler()

app.prepare().then(() => {
  createServer(async (req, res) => {
    try {

      const parsedUrl = parse(req.url, true)
      const { pathname, query } = parsedUrl

      if (pathname === '/a') {
        await app.render(req, res, '/a', query)
      } else if (pathname === '/b') {
        await app.render(req, res, '/b', query)
      } else {
        await handle(req, res, parsedUrl)
      }
    } catch (err) {
      console.error('Error occurred handling', req.url, err)
      res.statusCode = 500
      res.end('internal server error')
    }
  }).listen(port, (err) => {
    if (err) throw err
    console.log(`> Ready on http://${hostname}:${port}`)
  })
})

```

**Comment by himanshud360** [comment]
did you get the solution??

**Reply by hajpepe** [reply]
with nodejs no 
 exported as html :D

**Reply by hajpepe** [reply]
1 year ago nextjs version 12 now nextjs version 13 have same problem ?

**Reply by himanshud360** [reply]
still facing the same issue... try to resolve this mess... :(

**Reply by hajpepe** [reply]
if your app is landing page or static pages only export as html is good for backend application  don't know cause for backend apps i use vps or vercel

**Reply by himanshud360** [reply]
yup i have only static page website in node js. can you help me with the steps to export as html

**Reply by hajpepe** [reply]
html export doesn't support next image ( import Image from 'next/image' ) in nextjs 13 idk but in 12 doesn't  so
you need test or use external image imports 
and for export 
npm run export

**Reply by hajpepe** [reply]
after that  just Upload exported html/publicfolder into public_html

**Reply by himanshud360** [reply]
thanks man.. :)

**Reply by hajpepe** [reply]
You're welcome ^^ I hope helpful
